ibis/backends/tests/test_client.py

```diff
@@ -202,3 +203,263 @@ def test_separate_database(con, alternate_current_database, current_data_db):
     db = con.database(current_data_db)
     assert db.name == current_data_db
     assert tmp_db.name == alternate_current_database
+
+
+def create_temp_table_with_schema(con, temp_table_name, schema):
+
+    con.drop_table(temp_table_name, force=True)
+    con.create_table(temp_table_name, schema=schema)
+    temporary = con.table(temp_table_name)
+    assert len(temporary.execute()) == 0
+
+    return temporary
+
+
+@pytest.mark.only_on_backends(
+    ['sqlite', 'postgres', 'mysql'],
+    reason="run only if backend is SQLAlchemy based",
+)
+def test_insert_append_from_dataframe(con):
+    sch = ibis.schema(
+        [
+            ('first_name', 'string'),
+            ('last_name', 'string'),
+            ('department_name', 'string'),
+            ('salary', 'float64'),
+        ]
+    )
+
+    temp_table = 'temp_to_table'
+    temporary = create_temp_table_with_schema(con, temp_table, sch)
+
+    records = pd.DataFrame(
+        {
+            'first_name': ['X', 'Y', 'Z'],
+            'last_name': ['A', 'B', 'C'],
+            'department_name': ['XX', 'YY', 'ZZ'],
+            'salary': [400.0, 500.0, 600.0],
+        }
+    )
+
+    con.insert(temp_table, data_obj=records, if_exists='append')
+    assert len(temporary.execute()) == 3
+    tm.assert_frame_equal(temporary.execute(), records)
+
+
+@pytest.mark.only_on_backends(
+    ['sqlite', 'postgres', 'mysql'],
```

Comment:
Since we have now several tests, maybe we could define this in the top as a constant, like `SQLALCHEMY_BACKENDS = ['sqlite', 'postgres', 'mysql']`. Would make adding/removing backends much faster if needed in the future.

Reply:
Done!